lib/fog/compute/google/requests/insert_server.rb

```diff
@@ -129,12 +129,22 @@ def handle_networks(options)
           end
 
           # Objectify the network if needed
-          unless network.is_a? Network
+          unless network.is_a?(Network)
             network = networks.get(network)
+            network_interfaces = network.get_as_interface_config(access_config)
+          end
+
+          if options.key? "subnetwork"
+            subnetwork = options.delete "subnetwork"
+            # Objectify the subnetwork if needed
+            unless subnetwork.is_a?(Subnetwork)
+              subnetwork = subnetworks.get(subnetwork, region_name)
+              network_interfaces = subnetwork.update_interface_config(network_interfaces)
+            end
           end
 
           # Return a networkInterfaces array
-          [network.get_as_interface_config(access_config)]
+          [network_interfaces]
```

Comment:
Thank you for the replies.
Now this test passed well. 
lkuntar@lkuntar-VirtualBox:~/workspace/forks/fog-google$ rake test TEST="test/**/test_compute_networks_collection.rb"
Run options: --seed 48938

# Running:

.

Finished in 25.083174s, 0.0399 runs/s, 0.3189 assertions/s.

1 runs, 8 assertions, 0 failures, 0 errors, 0 skips